### ISSUE TITLE
Fix failing v2-beta entity subsumer test

### ISSUE THREAD

**Issue by johnbradley** [phenoscape/rphenoscape#212]
A entity subsumer test is failing when using the [v2-beta kb.phenoscape API](https://dev.phenoscape.org/api/v2-beta)
```
Failure (test-freqs.R:66:3): success rate for entity subsumer terms
max(type.fracs) is not strictly more than 0.9. Difference: -0.0231
```
https://github.com/phenoscape/rphenoscape/blob/82e16d96e3ed34c6ddd59ee640ce58310d684c62/tests/testthat/test-freqs.R#L61-L66
The v2-beta API is returning more phenotype items:
```
> table(tt.types)
tt.types
   entity phenotype   quality 
       57         7         1 
```
 With the production API:
```
> table(tt.types)
tt.types
   entity phenotype   quality 
       47         2         1 
```

**Comment by johnbradley** [comment]
I think the test is overly strict in this case and it's failing due to a change it the data.

**Comment by hlapp** [comment]
> I think the test is overly strict in this case and it's failing due to a change it the data.

I agree with the second part but not necessarily (i.e., before further investigation) with the first part. The test is relatively strict for a reason: It starts with collecting a few entity terms and then retrieves their subsumer terms. These should in theory _all_ be entity terms, and hence classify as such. If any, and certainly if a non-minuscule fraction, classify otherwise as quality or phenotype, that's a reason for concern, and review.

**Comment by johnbradley** [comment]
Perhaps the `term_category()` function is not correctly classifying these new `patterns` IRIs.
With the v2-beta API we end up with the following "phenotype" IRI:
```
> rownames(subs.mat)[tt.types == "phenotype"]
[1] "http://purl.org/phenoscape/kb/patterns/has_part.yaml#35fdeef4b6a61a37ea3d6fb1424521379d117dfe"
[2] "http://purl.org/phenoscape/kb/patterns/has_part.yaml#0fa773618d439e14c5d0c8454089b1429ba75b78"
[3] "http://purl.org/phenoscape/kb/patterns/has_part.yaml#4790142e0ee3c490b847afd96af664fd7de3b0d7"
[4] "http://purl.obolibrary.org/obo/BFO_0000002"                                                   
[5] "http://purl.org/phenoscape/kb/patterns/has_part.yaml#d67c8f91dc03b96f26680cf6e15610bab65b0a10"
[6] "http://purl.org/phenoscape/kb/patterns/has_part.yaml#79d6cd84dea8cebaf433e17c6d38dcb3f093bc49"
[7] "http://purl.org/phenoscape/kb/patterns/has_part.yaml#57c551e26482551cd7a7dd8fc325e076bdadae2f"
```
With the production API we end up with the following "phenotype" IRI:
```
> rownames(subs.mat)[tt.types == "phenotype"]
[1] "http://purl.obolibrary.org/obo/BFO_0000002"                                                                                                                            
[2] "http://purl.org/phenoscape/expression?value=%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000051%3E+some+%3Chttp%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FBFO_0000002%3E"
```

**Comment by johnbradley** [comment]
The [term_category()](https://github.com/phenoscape/rphenoscape/blob/82e16d96e3ed34c6ddd59ee640ce58310d684c62/R/term-type.R#L25) function determines the category of an IRI.
From a high level this function checks to see if a term is one of the known roots or has an ancestor that is one of the known roots. If that fails it runs term_category() on the top $superClass of the term. If no $superClasses found it tries to parse the IRI using [decode_entity_postcomp()](https://github.com/phenoscape/rphenoscape/blob/6b59eda736431a8f25e08969eebe2dcbbf3c3150/R/term-weights.R#L112-L122) looking for the word "some" in the IRI and returning the rels and entities found.
If no rels/entities found the function assumes the term must be "phenotype".

Obviously `decode_entity_postcomp()` cannot find a rel/entity in a IRI like "http://purl.org/phenoscape/kb/patterns/has_part.yaml#57c551e26482551cd7a7dd8fc325e076bdadae2f". So this will be a difference just due to the new url format.

**Comment by hlapp** [comment]
I'm wondering whether we should rewrite the `term_category()` function such that instead of trying to interpret IRIs (which necessarily depends on IRIs being interpretable, when in theory they should be opaque anyway) and using fallbacks, it will just return NA for input terms where it can't determine category with a reasonably high confidence.

Lists of terms for which we want to determine frequency should best be of one and the same category anyway, and the use case for when a user wouldn't necessarily know what this category is is kind of weak. It strikes me as one of those things of convenience that sounded nice to have at the time, but in hindsight it's not very clear what the realistic type of situation is where it would be asking a lot or too much of the user to simply know what the category of the terms in the list is.

Arguably, this should perhaps be turned into its own issue - I (or you) can do that if you agree.

**Comment by johnbradley** [comment]
I agree.

**Comment by johnbradley** [comment]
Fixed by #225
